algosdk/abi.py

```diff
@@ -0,0 +1,570 @@
+                # 8 Boolean values can be encoded into 1 byte
+                byte_length = self.static_length // 8
+                if self.static_length % 8 != 0:
+                    byte_length += 1
+                return byte_length
+            element_byte_length = self.child_type.byte_len()
+            return self.static_length * element_byte_length
+        elif self.abi_type_id == BaseType.Tuple:
+            size = 0
+            i = 0
+            while i < len(self.child_types):
+                if self.child_types[i].abi_type_id == BaseType.Bool:
+                    after = Type.find_bool(self.child_types, i, 1)
+                    i += after
+                    bool_num = after + 1
+                    size += bool_num // 8
+                    if bool_num % 8 != 0:
+                        size += 1
+                else:
+                    child_byte_size = self.child_types[i].byte_len()
+                    size += child_byte_size
+                i += 1
+            return size
+        else:
+            raise error.ABITypeError(
+                f"cannot get length of a dynamic type: {self.abi_type_id}"
+            )
+
+
+class Value:
+    """
+    Represents an ABI Value for encoding.
+
+    Args:
+        abi_type (Type): the type object of the ABI value.
+        value (Union[int, bool, str, bytes, list]): the value to be encoded.
+
+    Attributes:
+        abi_type (Type)
+        value (Union[int, bool, str, bytes, list])
+    """
+
+    def __init__(self, abi_type, val) -> None:
+        self.abi_type = abi_type
+        self.value = val
+
+    @staticmethod
+    def make_uint(value, size):
+        assert isinstance(value, int)
+        if value >= (2 ** size):
+            raise error.ABIValueError(
+                f"value {value} is too big to fit in size {size}"
+            )
+        return Value(Type.make_uint_type(size), value)
+
+    @staticmethod
+    def make_uint8(value):
+        return Value.make_uint(value, 8)
+
+    @staticmethod
+    def make_uint16(value):
+        return Value.make_uint(value, 16)
+
+    @staticmethod
+    def make_uint32(value):
+        return Value.make_uint(value, 32)
+
+    @staticmethod
+    def make_uint64(value):
+        return Value.make_uint(value, 64)
+
+    @staticmethod
+    def make_ufixed(value, size, precision):
+        assert isinstance(value, int)
+        ufixed_value_type = Type.make_ufixed_type(size, precision)
+        uint_value = Value.make_uint(value, size)
+        uint_value.abi_type = ufixed_value_type
+        return uint_value
+
+    @staticmethod
+    def make_bool(value):
+        assert isinstance(value, bool)
+        return Value(Type.make_bool_type, value)
+
+    @staticmethod
+    def make_string(value):
+        assert isinstance(value, str)
+        return Value(Type.make_string_type, value)
+
+    @staticmethod
+    def make_byte(value):
+        assert isinstance(value, bytes)
```

Comment:
`bytes` is actually the string byte array type. So you should probably check if the input is an int between 0 and 255 inclusive.